### PR TITLE
[KeyVault] updating examples to use ClientSecretCredential, fix parsing keyID

### PR DESCRIPTION
**Pull request by seankane-msft**
Addresses customer reported issue #16676

### CODE REVIEW

File: sdk/keyvault/azkeys/example_test.go [hunk 1]
```diff
@@ -21,7 +21,7 @@ var client *azkeys.Client
 
 func ExampleNewClient() {
 	vaultUrl := os.Getenv("AZURE_KEYVAULT_URL")
-	cred, err := azidentity.NewDefaultAzureCredential(nil)
+	cred, err := azidentity.NewClientSecretCredential("myTenantID", "myClientID", "myClientSecret", nil)
```
Comment (jhendrixMSFT) [inline sdk/keyvault/azkeys/example_test.go:24]: Why was this changed?  I'm not pushing back, but I thought that we wanted to promote `DefaultAzureCredential`.

File: sdk/keyvault/azkeys/crypto/crypto_client_test.go [hunk 2]
```diff
@@ -16,6 +16,20 @@ import (
 
 var ctx = context.TODO()
 
+func TestConstructor(t *testing.T) {
+	client, err := NewClient("https://fakekvurl.vault.azure.net/keys/key89075156/0b29f1d3760f4407aeb996868c9a02a7", &FakeCredential{}, nil)
+	require.NoError(t, err)
+	require.NotNil(t, client.kvClient)
+	require.Equal(t, client.keyID, "key89075156")
```
Comment (jhendrixMSFT) [inline sdk/keyvault/azkeys/crypto/crypto_client_test.go:23]: Is there a test that covers when key ID is missing?

Comment (seankane-msft) [reply sdk/keyvault/azkeys/crypto/crypto_client_test.go:23]: There is now

File: sdk/keyvault/azkeys/CHANGELOG.md [hunk 3]
```diff
@@ -7,6 +7,7 @@
 ### Breaking Changes
 
 ### Bugs Fixed
+* Fixes a bug in `crypto.NewClient` where the `keyID` was required in the path, it is no longer required but is recommended.
```
Comment (jhendrixMSFT) [inline sdk/keyvault/azkeys/CHANGELOG.md:10]: This says `keyID` but it looks like it's the `keyVersion` that's optional.

File: sdk/keyvault/azkeys/crypto/crypto_client.go [hunk 4]
```diff
@@ -57,17 +57,21 @@ func parseKeyIDAndVersion(id string) (string, string, error) {
 		return "", "", err
 	}
 
-	path := strings.Split(parsed.Path, "/")
+	if !strings.HasPrefix(parsed.Path, "/keys/") {
+		return "", "", fmt.Errorf("could not parse Key ID from %s", id)
```
Comment (jhendrixMSFT) [inline sdk/keyvault/azkeys/crypto/crypto_client.go:61]: I would change the error message here so it's different from line 67 (right now the message is ambiguous).

File: sdk/keyvault/internal/challenge_policy.go [hunk 5]
```diff
@@ -176,10 +176,38 @@ func (k KeyVaultChallengePolicy) getChallengeRequest(orig policy.Request) (*poli
...
+	return 0, nil
+}
+
+func (n nopCloser) Seek(offset int64, whence int) (int64, error) {
+	return 0, nil
+}
 
-	return req, err
+// NopCloser returns a ReadSeekCloser with a no-op close method wrapping the provided io.ReadSeeker.
+func NopCloser() io.ReadSeekCloser {
```
Comment (jhendrixMSFT) [inline sdk/keyvault/internal/challenge_policy.go:209]: This is in the `streaming` package.

https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/azcore/streaming/progress.go#L23

**Review by jhendrixMSFT — APPROVED** [review]
One minor thing to not copy the `nopCloser`
